### PR TITLE
add function check current driver rather than assuming default

### PR DESCRIPTION
**Pull request by mcgov**
Add some code to check which driver is currently bound rather than just assuming the default on a node.

### CODE REVIEW

File: lisa/nic.py [hunk 1]
```diff
@@ -158,6 +158,14 @@ def get_lower_nics(self) -> List[str]:
     def get_device_slots(self) -> List[str]:
         return [x.pci_slot for x in self.nics.values() if x.pci_slot]
 
+    def get_current_device_driver(self, nic: NicInfo) -> str:
+        cmd = f"readlink /sys/class/net/{nic.upper}/device/driver"
+        found_link = self._node.execute(cmd, expected_exit_code=0).stdout
+        driver_name = found_link.split("/")[-1]
```
Comment (squirrelsc) [inline lisa/nic.py]: Add an example above to help understanding the structure.

File: lisa/nic.py [hunk 2]
```diff
@@ -158,6 +158,14 @@ def get_lower_nics(self) -> List[str]:
...
+        assert_that(driver_name).is_not_equal_to("")
```
Comment (squirrelsc) [inline lisa/nic.py]: Add description to understand the error, if it happens.

File: lisa/nic.py [hunk 3]
```diff
@@ -158,6 +158,14 @@ def get_lower_nics(self) -> List[str]:
     def get_device_slots(self) -> List[str]:
         return [x.pci_slot for x in self.nics.values() if x.pci_slot]
 
+    def get_current_device_driver(self, nic: NicInfo) -> str:
```
Comment (squirrelsc) [inline lisa/nic.py]: 1. Pass in the nic_name is easier to use.
2. the name can be simple like `get_driver`.

File: lisa/nic.py (hunk 2 above)
Comment (mcgov) [reply lisa/nic.py]: I refactored to make it more clear that it's just checking for no output, switched to using PurePosixPath to emphasize we're grabbing the filename from the path.

Comment (squirrelsc) [reply lisa/nic.py]: The assert_that will raise an error message into the test result. The error message is only information showing in the PowerBI report. The target is to reduce the chance to check the code, and let user know what's wrong by this error message.

Comment (mcgov) [reply lisa/nic.py]: Ah, sloppy on my part. will fix

**Review by squirrelsc — DISMISSED** [review]

**Comment by mcgov** [comment]
had to resolve a conflict

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**